src/libostree/ostree-repo.c

```diff
@@ -3509,16 +3512,57 @@ reload_sysroot_config (OstreeRepo          *self,
    * https://github.com/ostreedev/ostree/issues/1719
    * https://github.com/ostreedev/ostree/issues/1801
    */
+  gboolean valid_bootloader;
   for (int i = 0; CFG_SYSROOT_BOOTLOADER_OPTS_STR[i]; i++)
     {
       if (g_str_equal (bootloader, CFG_SYSROOT_BOOTLOADER_OPTS_STR[i]))
         {
           self->bootloader = (OstreeCfgSysrootBootloaderOpt) i;
-          return TRUE;
+          valid_bootloader = TRUE;
+        }
+    }
+  if (!valid_bootloader) 
+  {
+    return glnx_throw (error, "Invalid bootloader configuration: '%s'", bootloader);
+  }
+  /* Parse bls-append-except-default string list. */
+  g_autoptr(GError) local_error = NULL;
+  g_auto(GStrv) read_values = g_key_file_get_string_list (self->config, "sysroot", "bls-append-except-default", NULL, &local_error);
+  /* We can ignore not found errors */
+  if (!read_values)
+    {
+      gboolean not_found = g_error_matches (local_error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_KEY_NOT_FOUND) || \
+                           g_error_matches (local_error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_GROUP_NOT_FOUND);
+      if (not_found)
+        {
+          g_clear_error (&local_error);
+        }
+      else
+        {
+          g_propagate_error (error, g_steal_pointer (&local_error));
+          return FALSE;
+        }
+    }
+  /* get all key value pairs in bls-append-except-default */
+  g_hash_table_unref (self->bls_append_values);
```

Comment:
It seems simpler to me to invoke [g_hash_table_remove_all](https://developer-old.gnome.org/glib/stable/glib-Hash-Tables.html#g-hash-table-remove-all) - then we don't have a period of time where `self->bls_append_values` is pointing to freed memory.